connman/src/inet.c

```diff
@@ -599,6 +599,16 @@ int connman_inet_add_network_route_with_metric(int index, const char *host,
 	addr.sin_addr.s_addr = inet_addr(host);
 	memcpy(&rt.rt_dst, &addr, sizeof(rt.rt_dst));
 
+	/*
+	 * Don't add a routes for link-local or unspecified
+	 * destination address coupled with unspecified gateway.
+	 */
+	if ((!host || is_addr_ll(AF_INET, (struct sockaddr *)&addr) || __connman_inet_is_any_addr(host, AF_INET))
```

Comment:
There is curious inconsistency between this function and ipv6 version. The latter null checks 'host' right at the start vs this one does not -> these added lines null check 'host' after it has already been used (ipv4) or when it has already been established that it is not null (ipv6).

Reply:
Dropped that patch. Let it simmer in upstream still. Does not seem to be urgent for us now.